src/storage/v2/inmemory/edge_type_index.cpp

```diff
@@ -195,6 +199,18 @@ void InMemoryEdgeTypeIndex::UpdateOnEdgeCreation(Vertex *from, Vertex *to, EdgeR
   acc.insert(Entry{from, to, edge_ref.ptr, tx.start_timestamp});
 }
 
+void InMemoryEdgeTypeIndex::UpdateOnEdgeModification(Vertex *old_from, Vertex *old_to, Vertex *new_from, Vertex *new_to,
+                                                     EdgeRef edge_ref, EdgeTypeId edge_type, const Transaction &tx) {
+  auto it = index_.find(edge_type);
+  if (it == index_.end()) {
+    return;
+  }
+  auto acc = it->second.access();
+
+  acc.remove(Entry{old_from, old_to, edge_ref.ptr, tx.start_timestamp});
```

Comment:
Does old version of edge has the same `tx.start_timestamp`? What it it was created in previous txs?

Reply:
It is always created in a previous one. Good catch. Fixed.